pkg/adapter/containers_remote.go

```diff
@@ -1031,3 +1038,243 @@ func (r *LocalRuntime) Commit(ctx context.Context, c *cliconfig.CommitValues, co
 func (r *LocalRuntime) Exec(c *cliconfig.ExecValues, cmd []string) error {
 	return define.ErrNotImplemented
 }
+
+func (r *LocalRuntime) parsePath(path string) (*Container, string) {
+	pathArr := strings.SplitN(path, ":", 2)
+	if len(pathArr) == 2 {
+		ctr, err := r.LookupContainer(pathArr[0])
+		if err == nil {
+			return ctr, pathArr[1]
+		}
+	}
+	return nil, path
+}
+
+func (r *LocalRuntime) CopyBetweenHostAndContainer(src string, dest string, extract, pause bool) error {
+
+	srcCtr, srcPath := r.parsePath(src)
+	destCtr, destPath := r.parsePath(dest)
+
+	if (srcCtr == nil && destCtr == nil) || (srcCtr != nil && destCtr != nil) {
+		return errors.Errorf("invalid arguments %s, %s you must use just one container", src, dest)
+	}
+
+	if len(srcPath) == 0 || len(destPath) == 0 {
+		return errors.Errorf("invalid arguments %s, %s you must specify paths", src, dest)
+	}
+	ctr := srcCtr
+	isFromHostToCtr := (ctr == nil)
+	if isFromHostToCtr {
+		ctr = destCtr
+	}
+
+	if isFromHostToCtr {
+
+		srcArr := strings.SplitN(src, ":", 2)
+		if len(srcArr) != 2 {
+			return errors.Errorf("must specify the location of source file, remote or local")
+		}
+		location := srcArr[0]
+		srcPath = srcArr[1]
+		if location == "" {
+			return errors.Errorf("must specify local or remote")
+		}
+		if location == "local" {
+			if strings.HasSuffix(dest, "/") {
+				destPath = fmt.Sprintf("%s/", destPath)
+			}
+			reply, err := iopodman.CopyTo().Send(r.Conn, varlink.Upgrade, location, srcPath, 0600, destPath, extract, pause, ctr.ID())
```

Comment:
Shouldn't mode match Src file vs. hard coded?

Reply:
the source file is from the remote file system this augument is not useful under this situation.